graphql_compiler/cost_estimation/helpers.py

```diff
@@ -1,33 +1,51 @@
 # Copyright 2019-present Kensho Technologies, LLC.
-from graphql import GraphQLInt
+from typing import Union, cast
+
+from graphql import (
+    GraphQLInt,
+    GraphQLInterfaceType,
+    GraphQLList,
+    GraphQLObjectType,
+    GraphQLScalarType,
+)
 
 from ..global_utils import is_same_type
 from ..schema import GraphQLDate, GraphQLDateTime
 from ..schema.schema_info import QueryPlanningSchemaInfo
 
 
+def _get_field_type(
+    schema_info: QueryPlanningSchemaInfo, vertex_name: str, field_name: str
+) -> Union[GraphQLList, GraphQLScalarType]:
```

Comment:
The return type here is (long-term) problematic as well. In principle, the type of a field can also be an interface, object, or union. In fact, some of the edge constraints are expressible in GraphQL SDL in ways that can cause this — for example, many-to-one edges are represented by `field_name: NeighborType` instead of `field_name: [NeighborType]`.

This hack is fine for now, but in the not-too-distant future, we should reconsider how we actually represent these ideas and make our approach as close as possible to vanilla GraphQL because that will improve our compatibility with the rest of the ecosystem.